### ISSUE TITLE
False Positive "ellipsis" is marked as @type_check_only and can be used only in type annotations

### ISSUE THREAD

**Issue by randolf-scholz** [microsoft/pyright#5954]
```python
from types import EllipsisType

x = ...
match x:
    case EllipsisType():  # ✘ error: "ellipsis" is marked as @type_check_only
        print(0)
```

**Comment by erictraut** [comment]
I think this is a bug in typeshed, not in pyright. The symbol `ellipsis` is decorated with `@type_check_only`, and `EllipsisType` is a type alias to the `ellipsis` symbol. Please file a bug or a PR with the typeshed maintainers.
